code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/TimerNotificationBuilder.kt

```diff
@@ -0,0 +1,188 @@
+import android.content.Intent
+import android.os.Build
+import android.os.SystemClock
+import android.view.View
+import android.widget.RemoteViews
+import androidx.core.app.NotificationCompat
+import com.adobe.marketing.mobile.notificationbuilder.NotificationConstructionFailedException
+import com.adobe.marketing.mobile.notificationbuilder.PushTemplateIntentConstants
+import com.adobe.marketing.mobile.notificationbuilder.R
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants.LOG_TAG
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateImageUtils
+import com.adobe.marketing.mobile.notificationbuilder.internal.extensions.createNotificationChannelIfRequired
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.TimerPushTemplate
+import com.adobe.marketing.mobile.services.Log
+
+private const val TAG = "TimerNotificationBuilder"
+
+/**
+ * Object responsible for constructing a [NotificationCompat.Builder] object containing a timer push template notification.
+ */
+internal object TimerNotificationBuilder {
+
+    data class TimerContent(
+        val title: String,
+        val body: String?,
+        val expandedBody: String?,
+        val imageUrl: String?
+    )
+
+    /**
+     * Constructs a notification for the timer push template
+     *
+     * @param context the context
+     * @param template the timer push template
+     * @param trackerActivityClass the tracker
+     * @param broadcastReceiverClass the broadcast receiver
+     * @return the notification builder
+     */
+    @Throws(NotificationConstructionFailedException::class)
+    fun construct(
+        context: Context,
+        template: TimerPushTemplate,
+        trackerActivityClass: Class<out Activity>?,
+        broadcastReceiverClass: Class<out BroadcastReceiver>?
+    ): NotificationCompat.Builder {
+
+        Log.trace(LOG_TAG, TAG, "Building a timer template push notification.")
+
+        val notificationManager =
+            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+
+        // create the notification channel if needed
+        val channelIdToUse = notificationManager.createNotificationChannelIfRequired(context, template)
+
+        // check if the template is expired
+        val isExpired = template.isExpired()
+        val (smallLayout, expandedLayout) = initializeLayouts(context, isExpired)
+
+        // create the notification builder with the common settings applied
+        val notificationBuilder = AEPPushNotificationBuilder.construct(
+            context,
+            template,
+            channelIdToUse,
+            trackerActivityClass,
+            smallLayout,
+            expandedLayout,
+            R.id.basic_expanded_layout
+        )
+
+        // create the timer content according to the expiry time
+        val timerContent = if (isExpired) {
```

Comment:
nit: Can we move `TimerContent` and this logic to `TimePushTemplate`? We have followed the same convention for other templates with data that changes like carousel and product rating